src/compile/layer.ts

```diff
@@ -71,8 +71,13 @@ export class LayerModel extends Model {
   }
 
   public parseSelection() {
-    // TODO: @arvind can write this
-    // We might need to split this into compileSelectionData and compileSelectionSignals?
+    this.component.selection = {};
+    this.children.forEach(child => {
+      child.parseSelection();
+      keys(child.component.selection).forEach((key) => {
```

Comment:
If I understand correctly, we always merge selection to the parent because selections exist across the entire hierarchy?  If so, maybe add a comment to explain.

I'm also curious why you don't need to remove the `selection` from `child.component`? Wouldn't this cause problem if you resolve and merge the selection later in the upper layers?  (I might have overlooked something, but probably a good idea to ask to clarify.)